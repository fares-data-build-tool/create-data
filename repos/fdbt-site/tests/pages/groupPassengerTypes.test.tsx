import * as React from 'react';
import { shallow } from 'enzyme';
import GroupPassengerTypes from '../../src/pages/groupPassengerTypes';

describe('pages', () => {
    describe('groupPassengerTypes', () => {
        it('should render correctly with no GroupPassengerTypes', () => {
            const tree = shallow(
                <GroupPassengerTypes
                    groupPassengerInfo={{
                        passengerTypes: [],
                    }}
                    csrfToken=""
                />,
            );
            expect(tree).toMatchSnapshot();
        });
        it('should render correctly with GroupPassengerTypes and errors (GroupPassengerTypesWithErrors)', () => {
            const tree = shallow(
                <GroupPassengerTypes
                    groupPassengerInfo={{
                        errors: [
                            {
                                errorMessage: 'Choose one or two passenger types - you cannot exceed this limit',
                                id: '',
                            },
                        ],
                    }}
                    csrfToken=""
                />,
            );
            expect(tree).toMatchSnapshot();
        });
        it('should render correctly with group info and no errors', () => {
            const tree = shallow(
                <GroupPassengerTypes
                    groupPassengerInfo={{
                        passengerTypes: ['adult', 'child'],
                    }}
                    csrfToken=""
                />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
