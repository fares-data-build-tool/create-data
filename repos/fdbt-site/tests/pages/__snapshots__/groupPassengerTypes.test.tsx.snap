// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages groupPassengerTypes should render correctly with GroupPassengerTypes and errors (GroupPassengerTypesWithErrors) 1`] = `
<Component
  description="Group Passengers selection page of the Fares Data Build Tool"
  errors={
    Array [
      Object {
        "errorMessage": "Choose one or two passenger types - you cannot exceed this limit",
        "id": "",
      },
    ]
  }
  title="Define Group Passengers - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/groupPassengerTypes"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Choose one or two passenger types - you cannot exceed this limit",
            "id": "",
          },
        ]
      }
    />
    <div
      className="govuk-form-group govuk-form-group--error"
    >
      <fieldset
        aria-describedby="passenger-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="passenger-type-page-heading"
          >
            Select a passenger type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="passenger-type-hint"
        >
          Select the passenger types included in your group ticket. Select the ‘Anyone’ option if your group contains more than two of the below choices.
        </span>
        <FormElementWrapper
          errorClass="govuk-checkboxes--error"
          errorId="passenger-type-0"
          errors={
            Array [
              Object {
                "errorMessage": "Choose one or two passenger types - you cannot exceed this limit",
                "id": "",
              },
            ]
          }
        >
          <div
            className="govuk-checkboxes"
          >
            <div
              className="govuk-checkboxes__item"
              key="adult"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-0"
                name="passengerTypes"
                type="checkbox"
                value="adult"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-0"
              >
                Adult
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="child"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-1"
                name="passengerTypes"
                type="checkbox"
                value="child"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-1"
              >
                Child
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="infant"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-2"
                name="passengerTypes"
                type="checkbox"
                value="infant"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-2"
              >
                Infant
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="senior"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-3"
                name="passengerTypes"
                type="checkbox"
                value="senior"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-3"
              >
                Senior
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="student"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-4"
                name="passengerTypes"
                type="checkbox"
                value="student"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-4"
              >
                Student
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="youngPerson"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-5"
                name="passengerTypes"
                type="checkbox"
                value="youngPerson"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-5"
              >
                Young Person
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="anyone"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-6"
                name="passengerTypes"
                type="checkbox"
                value="anyone"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-6"
              >
                Anyone
              </label>
            </div>
          </div>
        </FormElementWrapper>
      </fieldset>
    </div>
    <InsetText
      text="More passenger types will become available soon"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages groupPassengerTypes should render correctly with group info and no errors 1`] = `
<Component
  description="Group Passengers selection page of the Fares Data Build Tool"
  errors={Array []}
  title="Define Group Passengers - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/groupPassengerTypes"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="passenger-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="passenger-type-page-heading"
          >
            Select a passenger type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="passenger-type-hint"
        >
          Select the passenger types included in your group ticket. Select the ‘Anyone’ option if your group contains more than two of the below choices.
        </span>
        <FormElementWrapper
          errorClass="govuk-checkboxes--error"
          errorId="passenger-type-0"
          errors={Array []}
        >
          <div
            className="govuk-checkboxes"
          >
            <div
              className="govuk-checkboxes__item"
              key="adult"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-0"
                name="passengerTypes"
                type="checkbox"
                value="adult"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-0"
              >
                Adult
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="child"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-1"
                name="passengerTypes"
                type="checkbox"
                value="child"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-1"
              >
                Child
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="infant"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-2"
                name="passengerTypes"
                type="checkbox"
                value="infant"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-2"
              >
                Infant
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="senior"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-3"
                name="passengerTypes"
                type="checkbox"
                value="senior"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-3"
              >
                Senior
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="student"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-4"
                name="passengerTypes"
                type="checkbox"
                value="student"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-4"
              >
                Student
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="youngPerson"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-5"
                name="passengerTypes"
                type="checkbox"
                value="youngPerson"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-5"
              >
                Young Person
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="anyone"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-6"
                name="passengerTypes"
                type="checkbox"
                value="anyone"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-6"
              >
                Anyone
              </label>
            </div>
          </div>
        </FormElementWrapper>
      </fieldset>
    </div>
    <InsetText
      text="More passenger types will become available soon"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;

exports[`pages groupPassengerTypes should render correctly with no GroupPassengerTypes 1`] = `
<Component
  description="Group Passengers selection page of the Fares Data Build Tool"
  errors={Array []}
  title="Define Group Passengers - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/groupPassengerTypes"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="passenger-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="passenger-type-page-heading"
          >
            Select a passenger type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="passenger-type-hint"
        >
          Select the passenger types included in your group ticket. Select the ‘Anyone’ option if your group contains more than two of the below choices.
        </span>
        <FormElementWrapper
          errorClass="govuk-checkboxes--error"
          errorId="passenger-type-0"
          errors={Array []}
        >
          <div
            className="govuk-checkboxes"
          >
            <div
              className="govuk-checkboxes__item"
              key="adult"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-0"
                name="passengerTypes"
                type="checkbox"
                value="adult"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-0"
              >
                Adult
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="child"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-1"
                name="passengerTypes"
                type="checkbox"
                value="child"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-1"
              >
                Child
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="infant"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-2"
                name="passengerTypes"
                type="checkbox"
                value="infant"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-2"
              >
                Infant
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="senior"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-3"
                name="passengerTypes"
                type="checkbox"
                value="senior"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-3"
              >
                Senior
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="student"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-4"
                name="passengerTypes"
                type="checkbox"
                value="student"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-4"
              >
                Student
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="youngPerson"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-5"
                name="passengerTypes"
                type="checkbox"
                value="youngPerson"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-5"
              >
                Young Person
              </label>
            </div>
            <div
              className="govuk-checkboxes__item"
              key="anyone"
            >
              <input
                className="govuk-checkboxes__input"
                id="passenger-type-6"
                name="passengerTypes"
                type="checkbox"
                value="anyone"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="passenger-type-6"
              >
                Anyone
              </label>
            </div>
          </div>
        </FormElementWrapper>
      </fieldset>
    </div>
    <InsetText
      text="More passenger types will become available soon"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
